core/api_token_api.php

```diff
@@ -117,11 +124,15 @@ function api_token_validate( $p_username, $p_token ) {
 		return false;
 	}
 
-	$t_encrypted_token = api_token_hash( $p_token );
-
+	if( API_TOKEN_GENERIC == $p_kind ) {
+		$t_encrypted_token = api_token_hash( $p_token );
+	} else {
+		$t_encrypted_token = api_token_hash( $p_kind . $p_token );
```

Comment:
That is to avoid modification to places which does not take new concept of token type into account. E.g. just a safe net. 

If some old code would run a token check on restricted token - it will fail, because of kind addition. Otherwise you had to be extra sure that no other code checks tokens without also checking its type.

Yes, all token-related checks are supposed to go via api_token_validate (which takes type into account), but I would be better safe than sorry.